Jbarget/oxymore
# So/conscious shopping #9

# What changes have you made?
- added in the content to the conscious shopping page 
- Update/RD to myself. To explain what I'm trying to do: I've created an array of objects made up of the content that will change on click i.e. 2 images and 2 paragraphs of text 

Here I'm setting the starting image before the click event takes place. 
```
const startingImage = carousel[0].mainImg
const [current, setCurrent] = useState(startingImage);
```

To set the current image to the next image I then use the `setCurrent` hook to move to the next object in the array. But of course this doesn't work because the value of `current` is currently `startingImage` which is a string - and we can't add a number to a string. Even if the value of `current` was a number, we would need to add a `% carousel.length` in order that the index position never counts past the number of objects in the array. 

```
function handleClick() {
    setCurrent(current + 1);
  } 
```

What we need here is to set the initial state to the first object, provide the `handleClick()`some logic so that on click the current index increases by 1 and displays the next `mainImg`, `secondaryImg` and `interviewText`. When the index is higher than 5, then the index starts again from 0. 

- Update. I've now added functionality to the `onClick` so that the `mainImg`, `secondaryImg` and `interviewText` all change. 

- I've tried to implement a key down and key up function so that the user can toggle left and right on the carousel. I've added it inside the `handleClick()` and used `key` as the argument here but I think this might be where I'm going wrong.

- I've also created the project preview page with the timer in the same file as @Jbarget mentioned. This is quite a bit of code so not sure if it's a bit messy to read.. let me know if you want me to split it out into another file.

<img width="1440" alt="Screenshot 2020-10-19 at 23 49 38" src="https://user-images.githubusercontent.com/53219789/96515693-0f719c00-1266-11eb-94d6-8488e6e8d7e6.png">

### Which issue(s) does this PR fix?
#9 

### Screenshots (if there are design changes)
<img width="223" alt="Screenshot 2020-10-12 at 19 29 14" src="https://user-images.githubusercontent.com/53219789/95775156-293d3d00-0cc2-11eb-8f58-472659e3e458.png">

<img width="1438" alt="Screenshot 2020-10-12 at 19 34 48" src="https://user-images.githubusercontent.com/53219789/95775098-06ab2400-0cc2-11eb-99b9-252f943182dc.png">


### How to test
- Click on the main image and check that both images change as well as the two paragraphs
- Make sure that the translation functionality still works
